### PR TITLE
chore(finalizer): add retry to finalizer

### PR DESCRIPTION
**Pull request by slonka**
Coming from service review.

Do we want to make this configurable?

### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues --
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by michaelbeaumont — APPROVED** [review]
For some reason I thought we had this and it was happening in spite of the retry. Nice

**Comment by michaelbeaumont** [comment]
> Do we want to make this configurable?

We have retries elsewhere on insights that we haven't had reason to configure yet. I don't see a reason to start with this one.

**Review by lahabana — APPROVED** [review]

File: pkg/gc/finalizer.go [hunk 1]
```diff
@@ -181,7 +181,7 @@ func (f *subscriptionFinalizer) checkGeneration(ctx context.Context, typ core_mo
 				}
 			}
 			return nil
-		}); err != nil {
+		}, manager.WithConflictRetry(500*time.Millisecond, 3, 30)); err != nil {
```
[the remark this answers is not in the document]
Comment (slonka) [reply pkg/gc/finalizer.go]: added

**Review by jakubdyszkiewicz — APPROVED** [review]

File: pkg/gc/finalizer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/gc/finalizer.go]: ok, good to know!

**Pull request merged**